python/src/nanoarrow/schema.py

```diff
@@ -0,0 +1,448 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import enum
+from typing import Union
+
+from nanoarrow._lib import CArrowTimeUnit, CArrowType, CSchemaBuilder, CSchemaView
+from nanoarrow.c_lib import c_schema
+
+
+class Type(enum.Enum):
+    """The Type enumerator provides a means by which the various type
+    categories can be identified. Type values can be used in place of
+    :class:`Schema` instances in most places for parameter-free types."""
+
+    UNINITIALIZED = CArrowType.UNINITIALIZED
+    NA = CArrowType.NA
+    BOOL = CArrowType.BOOL
+    UINT8 = CArrowType.UINT8
+    INT8 = CArrowType.INT8
+    UINT16 = CArrowType.UINT16
+    INT16 = CArrowType.INT16
+    UINT32 = CArrowType.UINT32
+    INT32 = CArrowType.INT32
+    UINT64 = CArrowType.UINT64
+    INT64 = CArrowType.INT64
+    HALF_FLOAT = CArrowType.HALF_FLOAT
+    FLOAT = CArrowType.FLOAT
+    DOUBLE = CArrowType.DOUBLE
+    STRING = CArrowType.STRING
+    BINARY = CArrowType.BINARY
+    FIXED_SIZE_BINARY = CArrowType.FIXED_SIZE_BINARY
+    DATE32 = CArrowType.DATE32
+    DATE64 = CArrowType.DATE64
+    TIMESTAMP = CArrowType.TIMESTAMP
+    TIME32 = CArrowType.TIME32
+    TIME64 = CArrowType.TIME64
+    INTERVAL_MONTHS = CArrowType.INTERVAL_MONTHS
+    INTERVAL_DAY_TIME = CArrowType.INTERVAL_DAY_TIME
+    DECIMAL128 = CArrowType.DECIMAL128
+    DECIMAL256 = CArrowType.DECIMAL256
+    LIST = CArrowType.LIST
+    STRUCT = CArrowType.STRUCT
+    SPARSE_UNION = CArrowType.SPARSE_UNION
+    DENSE_UNION = CArrowType.DENSE_UNION
+    DICTIONARY = CArrowType.DICTIONARY
+    MAP = CArrowType.MAP
+    EXTENSION = CArrowType.EXTENSION
+    FIXED_SIZE_LIST = CArrowType.FIXED_SIZE_LIST
+    DURATION = CArrowType.DURATION
+    LARGE_STRING = CArrowType.LARGE_STRING
+    LARGE_BINARY = CArrowType.LARGE_BINARY
+    LARGE_LIST = CArrowType.LARGE_LIST
+    INTERVAL_MONTH_DAY_NANO = CArrowType.INTERVAL_MONTH_DAY_NANO
+
+    def __arrow_c_schema__(self):
+        # This will only work for parameter-free types
+        c_schema = CSchemaBuilder.allocate().set_type(self.value).finish()
+        return c_schema._capsule
```

Comment:
Is this needed? (given it does not always work)
Or this ensures you can pass it directly to `c_schema`? (but that is not necessarily something that a user needs to do?)